app/src/main/java/fr/free/nrw/commons/upload/worker/UploadWorker.kt

```diff
@@ -180,12 +182,16 @@ class UploadWorker(var appContext: Context, workerParams: WorkerParameters) :
                         if (contribution.state == Contribution.STATE_QUEUED) {
                             contribution.state = Contribution.STATE_QUEUED_LIMITED_CONNECTION_MODE
                             contributionDao.save(contribution)
+                        } else {
+                            //To Nothing
                         }
                     } else {
                         contribution.transferred = 0
                         contribution.state = Contribution.STATE_IN_PROGRESS
                         contributionDao.save(contribution)
                         uploadContribution(contribution = contribution)
+                        setProgressAsync(Data.Builder().putInt("progress", countUpload).build())
```

Comment:
Please add a comment explaining what this line does.

Reply:
This function updates the liveData when media is uploaded